persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideImpl.scala

```diff
@@ -62,65 +62,52 @@ private[lagom] class ReadSideImpl @Inject()(
       clazz: Class[_]
   ) = {
 
-    // Only run if we're configured to run on this role
-    if (config.role.forall(Cluster(system).selfRoles.contains)) {
-      // try to create one instance to fail fast (e.g. wrong constructor)
-      val dummyProcessor = try {
-        processorFactory()
-      } catch {
-        case NonFatal(e) =>
-          throw new IllegalArgumentException(
-            "Cannot create instance of " +
-              s"[${clazz.getName}]",
-            e
```

Comment:
We are loosing information here. The fail-fast message was at least being more explicit towards the user.

Reply:
Remember to keep the `cause` when fixing this.

Reply:
Good call. I think the sync'ing will have to be in the opposite direction then because the `scaladsl` version didn't have the failfast safeguard.

Reply:
> Remember to keep the cause when fixing this.

WDYM? The cause is kept in `e` (second argument to `IllegalArgumentException`), right?

Reply:
> the sync'ing will have to be in the opposite direction

Not true. This check is only required in `javadsl` because we're using runtime DI, while in `scaladsl` we use compile-time DI.

Reply:
In any case, since this failfast check was created until today, the rest of the code in this class has changed so much that this is no longer a `failfast` verification but a way to provide a more user-friendly error message. 

Originally, the constructor exercised here was not used until quite later in the process: it was passed to some underlying layers (the cluster startup task of the projection worker) in which cases debugging the failure was harder. In https://github.com/lagom/lagom/pull/1117 the constructor was used in `ReadSideImpl` for the first time using it earlier than before. 

So, yeah, the `try{}catch` is useful but I've adapted the comment.